Classes/WPEditorFormatbarView.m

```diff
@@ -0,0 +1,528 @@
+        WPEditorToolbarButton* htmlButton = (WPEditorToolbarButton*)self.htmlButton.customView;
+        NSAssert([htmlButton isKindOfClass:[WPEditorToolbarButton class]],
+                 @"Expected to have an HTML button of class WPEditorToolbarButton here.");
+        
+        htmlButton.selectedTintColor = _selectedItemTintColor;
+    }
+}
+
+#pragma mark - Toolbar items
+
+- (void)enableToolbarItems:(BOOL)enable
+    shouldShowSourceButton:(BOOL)showSource
+{
+    for (ZSSBarButtonItem *item in self.leftToolbar.items) {
+        if (item.tag == kWPEditorViewControllerElementShowSourceBarButton) {
+            item.enabled = showSource;
+        } else {
+            item.enabled = enable;
+            
+            if (!enable) {
+                [item setSelected:NO];
+            }
+        }
+    }
+    
+    for (ZSSBarButtonItem *item in self.regularToolbar.items) {
+        if (item.tag == kWPEditorViewControllerElementShowSourceBarButton) {
+            item.enabled = showSource;
+        } else {
+            item.enabled = enable;
+            
+            if (!enable) {
+                [item setSelected:NO];
+            }
+        }
+    }
+}
+
+- (void)clearSelectedToolbarItems
+{
+    for (ZSSBarButtonItem *item in self.leftToolbar.items) {
+        if (item.tag != kWPEditorViewControllerElementShowSourceBarButton) {
+            [item setSelected:NO];
+        }
+    }
+    
+    for (ZSSBarButtonItem *item in self.regularToolbar.items) {
+        if (item.tag != kWPEditorViewControllerElementShowSourceBarButton) {
+            [item setSelected:NO];
+        }
+    }
+}
+
+- (void)selectToolbarItemsForStyles:(NSArray*)styles
+{
+    for (UIBarButtonItem *item in self.leftToolbar.items) {
+        if ([item isKindOfClass:[ZSSBarButtonItem class]]) {
+            ZSSBarButtonItem* zssItem = (ZSSBarButtonItem*)item;
+            
+            if ([styles containsObject:zssItem.htmlProperty]) {
+                zssItem.selected = YES;
+            } else {
+                zssItem.selected = NO;
+            }
+        }
+    }
+    
+    for (UIBarButtonItem *item in self.regularToolbar.items) {
+        if ([item isKindOfClass:[ZSSBarButtonItem class]]) {
+            ZSSBarButtonItem* zssItem = (ZSSBarButtonItem*)item;
+            
+            if ([styles containsObject:zssItem.htmlProperty]) {
+                zssItem.selected = YES;
+            } else {
+                zssItem.selected = NO;
+            }
+        }
+    }
+}
+
+#pragma mark - Toolbar: buttons
+
+- (void)initBarButtonItem:(ZSSBarButtonItem*)barButtonItem
+                  withTag:(WPEditorViewControllerElementTag)tag
+             htmlProperty:(NSString*)htmlProperty
+                imageName:(NSString*)imageName
+                   target:(id)target
+                 selector:(SEL)selector
+       accessibilityLabel:(NSString*)accessibilityLabel
+{
+    NSAssert(barButtonItem, @"ZSSBarButtonItem should not be nil here.");
```

Comment:
I like using `NSParameterAssert` but I think that's just personal preference not a rule. :smile: